test/SafexMain.js

```diff
@@ -0,0 +1,255 @@
+
+      // FAILURE : metaEvidence is not passed
+      await expect(
+        safexMain.connect(acc1).createPlan(acc8.address, "", {
+          value: arbitrationFee.toNumber(),
+        })
+      ).to.be.revertedWith("Should provide metaEvidence to create a plan");
+
+      // FAILURE : inheritor is an zero address
+      await expect(
+        safexMain
+          .connect(acc1)
+          .createPlan(
+            "0x0000000000000000000000000000000000000000",
+            "https://bafybeif52vrffdp7m2ip5f44ox552r7p477druj2w4g3r47wpuzdn7235y.ipfs.infura-ipfs.io/",
+            {
+              value: arbitrationFee.toNumber(),
+            }
+          )
+      ).to.be.revertedWith("Should provide an inheritor for the plan");
+
+      // FAILURE : plan creator and inheritor are same
+      await expect(
+        safexMain
+          .connect(acc1)
+          .createPlan(
+            acc1.address,
+            "https://bafybeif52vrffdp7m2ip5f44ox552r7p477druj2w4g3r47wpuzdn7235y.ipfs.infura-ipfs.io/",
+            {
+              value: arbitrationFee.toNumber(),
+            }
+          )
+      ).to.be.revertedWith("Plan creator should not be the inheritor of the plan");
+    });
+
+    it("Should allow inheritors to create a claim", async () => {
+      expect(await safexMain.getSafexMainContractBalance()).to.equal(2000000000000000); // 0.002 eth
+
+      let plan;
+      plan = await safexMain.plans(1);
+
+      expect(plan.planFunds).to.equal(2000000000000000); // 0.002 eth
+
+      // FAILURE : plan does not exist
+      await expect(
+        safexMain
+          .connect(acc9)
+          .createClaim(4, "https://bafybeif52vrffdp7m2ip5f44ox552r7p477druj2w4g3r47wpuzdn7235y.ipfs.infura-ipfs.io/")
+      ).to.be.revertedWith("Plan does not exist");
+
+      // FAILURE : only inheritor of the plan can create the claim
+      await expect(
+        safexMain
+          .connect(acc8)
+          .createClaim(1, "https://bafybeif52vrffdp7m2ip5f44ox552r7p477druj2w4g3r47wpuzdn7235y.ipfs.infura-ipfs.io/")
+      ).to.be.revertedWith("Only inheritor of the plan can create the claim");
+
+      // SUCCESS : create 1st claim
+      await safexMain
+        .connect(acc9)
+        .createClaim(1, "https://bafybeif52vrffdp7m2ip5f44ox552r7p477druj2w4g3r47wpuzdn7235y.ipfs.infura-ipfs.io/"); // evidence
+
+      expect(await safexMain.getSafexMainContractBalance()).to.equal(1000000000000000); // 0.001 eth
+
+      plan = await safexMain.plans(1);
+
+      expect(plan.planFunds).to.equal(1000000000000000); // 0.001 eth
+      expect(plan.claimsCount).to.equal(1);
+
+      const claim1 = await safexMain.claims(0);
+
+      expect(claim1.planId).to.equal(1);
+      expect(claim1.disputeId).to.equal(0);
+      expect(claim1.claimedBy).to.equal(acc9.address);
+      expect(claim1.metaEvidenceId).to.equal(1);
+      expect(claim1.evidenceGroupId).to.equal(1);
+      expect(claim1.status).to.equal(0); // Active
+      expect(claim1.result).to.equal("Active");
+
```

Comment:
Don't check if all the values are equal. Check only the important ones like address and result. Try doing the same for all other places.